crates/core/src/contracts/contract_account.rs

```diff
@@ -29,7 +29,10 @@ impl<'a, P: Provider + Send + Sync> ContractAccount<'a, P> {
             ProviderError::StarknetError(starknet_error) => match starknet_error {
                 // TODO: we just need to test against ContractNotFound but madara is currently returning the wrong
                 // error See https://github.com/keep-starknet-strange/madara/issues/853
-                StarknetError::ContractError | StarknetError::ContractNotFound => Ok(vec![]),
+                StarknetError::ContractError | StarknetError::ContractNotFound => {
+                    log::error!("error in provider.call: {:?}", err);
```

Comment:
why do we log this? btw the madara issue has been closed so I've created an issue for use to remove the extra match arg

Reply:
I had an long debug session where the error ended being that I deployed the incorrect class hash. However, I wasn't getting any feedback from the function, just an empty bytecode due to the match. Figured adding the log would help debugging

Reply:
let's remove the log in the newly created issue for skimming the match